compiler/next/lib/resolution/resolution-types.cpp

```diff
@@ -17,11 +17,235 @@
  * limitations under the License.
  */
 
+#include "chpl/queries/update-functions.h"
 #include "chpl/resolution/resolution-types.h"
+#include "chpl/uast/Builder.h"
+#include "chpl/uast/Formal.h"
 
 namespace chpl {
 namespace resolution {
 
+using namespace uast;
+
+void ResolutionResultByPostorderID::setupForSymbol(const ASTNode* ast) {
+  assert(Builder::astTagIndicatesNewIdScope(ast->tag()));
+  vec.resize(ast->id().numContainedChildren());
+
+  symbolId = ast->id();
+}
+void ResolutionResultByPostorderID::setupForSignature(const Function* func) {
+  int bodyPostorder = 0;
+  if (func && func->body())
+    bodyPostorder = func->body()->id().postOrderId();
+  assert(0 <= bodyPostorder);
+  vec.resize(bodyPostorder);
+
+  symbolId = func->id();
+}
+void ResolutionResultByPostorderID::setupForFunction(const Function* func) {
+  setupForSymbol(func);
+}
+
+bool ResolutionResultByPostorderID::update(ResolutionResultByPostorderID& keep,
+                                           ResolutionResultByPostorderID& addin)
+{
+  return defaultUpdateVec(keep.vec, addin.vec);
+}
+
+bool FormalActualMap::computeAlignment(const UntypedFnSignature* untyped,
+                                       const TypedFnSignature* typed,
+                                       const CallInfo& call) {
+
+  // untyped must be provided but typed can be nullptr.
+  assert(untyped);
+
+  // create the mapping handling named and default arguments
+
+  // clear the current mapping
+  byFormalIdx.empty();
+  actualIdxToFormalIdx.empty();
+  mappingIsValid = false;
+  failingActualIdx = -1;
+
+  // allocate space in the arrays
+  byFormalIdx.resize(untyped->formals.size());
+  actualIdxToFormalIdx.resize(call.actuals.size());
+
+  // initialize the FormalActual parts from the Formals
+  size_t formalIdx = 0;
+  for (const Formal* formal : untyped->formals) {
+    FormalActual& entry = byFormalIdx[formalIdx];
+    entry.formal = formal;
+    if (typed) {
+      entry.formalType = typed->formalTypes[formalIdx];
+    }
+    entry.hasActual = false;
```

Comment:
We just cleared previous version of entry - shouldn't this be the default value for a new one?  Ditto `actualIdx`

Reply:
I'm not following. Are you saying we don't need this line? Sometimes I think it's worth having redundant code (e.g. here setting a bool) if it makes it clearer what is going on.